Azure/azure-functions-dotnet-worker
# Add extension validation step to CI

### Issue describing the changes in this PR

resolves https://github.com/Azure/azure-functions-dotnet-worker/issues/2237

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

This PR introduces a target imported by each extension library that will generate a `.csproj` file referencing their respective WebJobs extension package. This will allow CI to build and restore these packages, producing a `project.assets.json` file that can be scanned by Component Governance. Full context in the linked issue.

#### Important Change for Contributors

WebJobs extensions would no longer be referenced in `Properties/AssemblyInfo.cs` - they have been moved to the `.csproj` file of each extension as an MS Build property. To update the WebJobs extension package, please update this new MS Build property, `WebJobsExtension`.

#### Testing Locally

1. Switch to this branch
2. Go into any extension under `/extensions` and build the project.
3. Navigate to `bin/Debug/net8.0/ExtensionValidation` to see the generated file.
